renderdoc/driver/vulkan/vk_core.cpp

```diff
@@ -705,6 +705,59 @@ void WrappedVulkan::SetDebugMessageSink(WrappedVulkan::ScopedDebugMessageSink *s
   Threading::SetTLSValue(debugMessageSinkTLSSlot, (void *)sink);
 }
 
+void WrappedVulkan::AddPendingCommandBufferCallbacks(VkCommandBuffer commandBuffer)
+{
+  VkResourceRecord *cmdRecord = GetRecord(commandBuffer);
+
+  rdcarray<std::function<void()>> callbacks;
+  {
+    SCOPED_LOCK(m_PendingCmdBufferCallbacksLock);
```

Comment:
I don't think you need to lock `m_PendingCmdBufferCallbacksLock` here? It should only be necessary below to guard `m_PendingCmdBufferCallbacks`.

Reply:
Not it's not, I'll remove it.  Though will accesses to `cmdRecord->...->pendingSubmissionCompleteCallbacks` need synchronising?  I suspose it does as additions to it are triggered by `vkCmd*` calls which can be called from mutliple threads.

Reply:
That should be safe based on the vulkan rules - only one thread can be recording to a command buffer at any given time.